packages/test/test-end-to-end-tests/src/test/offline/blobHandles.spec.ts

```diff
@@ -0,0 +1,144 @@
+import {
+	type ITestObjectProvider,
+	createTestConfigProvider,
+} from "@fluidframework/test-utils/internal";
+
+const interceptResult = <T>(
+	parent: any,
+	fn: (...args: any[]) => Promise<T>,
+	intercept: (result: T) => void | Promise<void>,
+) => {
+	const interceptFn = async (...args: any[]) => {
+		const val = await fn.apply(parent, args);
+		await intercept(val);
+		return val as T;
+	};
+	parent[fn.name] = interceptFn;
+	interceptFn.bind(parent);
+	return fn;
+};
+
+describeCompat("Odsp Network calls", "NoCompat", (getTestObjectProvider, apis) => {
+	const { DataObjectFactory, DataObject } = apis.dataRuntime;
+	const { ContainerRuntimeFactoryWithDefaultDataStore } = apis.containerRuntime;
+
+	// A Test Data Object that exposes some basic functionality.
+	class TestDataObject extends DataObject {
+		public get _root() {
+			return this.root;
+		}
+		public get _runtime() {
+			return this.runtime;
+		}
+		public handleGetPromise: Deferred<Promise<unknown>> = new Deferred();
+		protected async hasInitialized() {
+			// This is needed to first process the handle op before the blob attach op is processed
+			this._root.on("valueChanged", (changed) => {
+				const value = this._root.get<IFluidHandle>(changed.key);
+				assert(value !== undefined, "Value should exist");
+				this.handleGetPromise.resolve(value.get());
+			});
+		}
+	}
+	// Allow us to control summaries
+	const runtimeOptions: IContainerRuntimeOptions = {
+		summaryOptions: {
+			summaryConfigOverrides: {
+				state: "disabled",
+			},
+		},
+	};
+	const configProvider = createTestConfigProvider({
+		"Fluid.Container.enableOfflineLoad": true,
+	});
+	const testDataObjectType = "TestDataObject";
+	const dataObjectFactory = new DataObjectFactory(testDataObjectType, TestDataObject, [], {});
+
+	// The 1st runtime factory, V1 of the code
+	const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore({
+		defaultFactory: dataObjectFactory,
+		registryEntries: [dataObjectFactory.registryEntry],
+		runtimeOptions,
+	});
+
+	let provider: ITestObjectProvider;
+	let deferred: Deferred<void>;
+	let throwOnBlobCreate = false;
+
+	beforeEach("setup", async function () {
+		provider = getTestObjectProvider();
+		deferred = new Deferred<void>();
+		const documentServiceFactory = provider.documentServiceFactory;
+		interceptResult(
+			documentServiceFactory,
+			documentServiceFactory.createDocumentService,
+			(documentService) => {
+				interceptResult(documentService, documentService.connectToStorage, (storage) => {
+					interceptResult(storage, storage.createBlob, async (blob) => {
+						await deferred.promise;
+						if (throwOnBlobCreate) {
+							throwOnBlobCreate = false;
+							const error: any = new Error("Blob fetch failed");
+							error.canRetry = true;
+							console.log("Throwing error");
+							throw error;
+						}
+					});
+				});
+			},
+		);
+	});
+
+	it("Should have blob when handle op is sequenced", async () => {
```

Comment:
>it [](http://example.com/codeflow?start=1&length=2)

can we annotate this multi-step test, possibly leveraging what you already have in PR Description. Helps readability.